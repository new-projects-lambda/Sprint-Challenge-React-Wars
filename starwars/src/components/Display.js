import React from 'react'
import styled from 'styled-components';

const StyledDiv = styled.div`
    background: #cccccc99;
    width: 50%;
    color: #335577;
    margin-bottom: 30px;
    padding-left: 15px;
    
`;

export default function Display(props) {
    return (
        <StyledDiv>
            <h2>{props.data.name}</h2>
        </StyledDiv>
    )
}
