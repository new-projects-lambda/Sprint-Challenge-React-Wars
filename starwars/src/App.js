import React, { useState, useEffect }from 'react';
import './App.css';
import axios from 'axios';
import styled from 'styled-components';
import Display from './components/Display';

const App = () => {
  // Try to think through what state you'll need for this app before starting. Then build out
  // the state properties here.

  // Fetch characters from the star wars api in an effect hook. Remember, anytime you have a 
  // side effect in a component, you want to think about which state and/or props it should
  // sync up with, if any.
  const [data, setData] = useState();
  const StyledApp = styled.div`
    background: #00000099;
    display: flex;
    flex-direction: column;
    align-items: center;
  `;
  const Container = styled.div`
    width: 80%;
    margin: 30px auto;
  `;
  const StyledH1 = styled.h1`
    color: #4488aa;
  `;
  const fetchData = () => {
    axios.get(
      'https://swapi.co/api/people/'
    )
      .then(res => {
        setData(res.data.results)
      })
      .catch(err => console.log(err));
  };

  useEffect(() => {
    fetchData();
  }, []);
  if (!data) return <h3>Loading Data</h3>;
  return (
    <Container>
      <StyledApp>
        <StyledH1 className="Header">React Wars</StyledH1>
        {data.map(res => <Display data={res} key={res.name} />)}
      </StyledApp>
    </Container>
  );
}

export default App;
