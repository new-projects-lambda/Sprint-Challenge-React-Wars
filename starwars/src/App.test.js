
import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import Display from './components/Display';

import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});




describe('<App />', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<App />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('<Display />', () => {
  it('should match snapshot', () => {
    const tree2 = renderer.create(<Display />).toJSON();

    expect(tree2).toMatchSnapshot();
  });
});