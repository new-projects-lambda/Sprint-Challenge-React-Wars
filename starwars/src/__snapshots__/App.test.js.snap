// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
<h3>
  Loading Data
</h3>
`;
